### ISSUE TITLE
What about MC-G01?

### ISSUE THREAD

**Issue by michbeck100** [wangyu-/canon_mc-g02_resetter#2]
Does this work with MC-G01, too?

**Comment by raichea** [comment]
I've been looking into this and can see no reason why it wouldn't work for MC-G01 - it uses the same I2C EEPROM and the board looks identical.

One thing was puzzling me. The board appears to have a resistor pulling pin 7 up to Vcc. Pin 7 is WC - write control - same as write protect for other versions of the chip - and when high inhibits writing to the chip. There's no contact connected to pin 7 to pull it to GND to enable writing but the printer obviously does write to the chip. Checking with a multi-meter, however, the 
SMD component appears to be open circuit and pin 7 is connected to GND on the board, so the part is not a resistor, it's a capacitor that is simply placed across the supply rails.

**Comment by gaby64** [comment]
Anyone has a dump of MC-G01?

**Comment by gaby64** [comment]
> I've been looking into this and can see no reason why it wouldn't work for MC-G01 - it uses the same I2C EEPROM and the board looks identical.
> 
> One thing was puzzling me. The board appears to have a resistor pulling pin 7 up to Vcc. Pin 7 is WC - write control - same as write protect for other versions of the chip - and when high inhibits writing to the chip. There's no contact connected to pin 7 to pull it to GND to enable writing but the printer obviously does write to the chip. Checking with a multi-meter, however, the SMD component appears to be open circuit and pin 7 is connected to GND on the board, so the part is not a resistor, it's a capacitor that is simply placed across the supply rails.

Is it possible to hex edit the rom to change the counter?
Or can you send me a MC-G01 dump?

Thanks

**Comment by raichea** [comment]
> Is it possible to hex edit the rom to change the counter?

As described in the Wiki

> Or can you send me a MC-G01 dump?

Sorry, no - for the reasons given by the author in the Wiki

**Comment by gaby64** [comment]
> > Is it possible to hex edit the rom to change the counter?
> 
> As described in the Wiki
> 
> > Or can you send me a MC-G01 dump?
> 
> Sorry, no - for the reasons given by the author in the Wiki

That was a reason for not uploading on github, sharing between users is fine.
Pastebin is a nice place for such things if one wishes to stay anonymous.

**Comment by wangyu-** [comment]
Be aware before sharing the rom with people you don't know.

Previously, there was a person keep trying to ask the rom from me. Later it turns out he was planning to build customer resetter and plan to sell it.

**Comment by gaby64** [comment]
Has anyone analyzed the contents of the rom? Or a diff of dumps before and after count increment? Are we dealing with crypto signatures or encrypted content? Is it different from one unit to the other?
Given how many knockoff chips and resetter devices there are, I doubt there is any traceability or fingerprinting.

**Comment by wangyu-** [comment]
>Has anyone analyzed the contents of the rom? Or a diff of dumps before and after count increment? Are we dealing with crypto signatures or encrypted content? Is it different from one unit to the other?

For mc-g02, after you use the cartridge once,  almost all the chars inside the rom get changed. I couldn't find an obvious counter inside it.

Not sure if they use some encryption (and I believe no). But there seems to be some encoding/decoding used, so that you cannot see the raw counters.

**Comment by raichea** [comment]
> For mc-g02, after you use the cartridge once, almost all the chars inside the rom get changed. I couldn't find an obvious counter inside it.
> 
> Not sure if they use some encryption (and I believe no). But there seems to be some encoding/decoding used, so that you cannot see the raw counters.

Exactly - it's not just a simple counter.  The cartridge serial number is in the chip as well.

**Comment by raichea** [comment]
> Be aware before sharing the rom with people you don't know.

I agree

> Previously, there was a person keep trying to ask the rom from me. Later it turns out he was planning to build customer resetter and plan to sell it.

If someone is that interested in getting the contents, buying a new cartridge isn't that expensive.

**Comment by gaby64** [comment]
Anyone know abiut the Canon Service Tool?

https://resetter.net/canon-service-tool-5610

**Comment by gaby64** [comment]
Look what someone was nice enough to post

https://pastebin.com/z24TQ36x

**Comment by gaby64** [comment]
Look what someone was nice enough to post

https://pastebin.com/z24TQ36x

**Comment by gaby64** [comment]
![IMG_20231014_014054](https://github.com/wangyu-/canon_mc-g02_resetter/assets/2108555/a621d0b3-0a18-47e3-8d98-f5a687c3bbe7)

**Comment by wangyu-** [comment]
@gaby64 Does the rom in pastebin reset your old mc-g01 successfully?

By the way your rom looks very different from my mc-g02's. Yours contains far more zeros.

**Comment by raichea** [comment]
I've looked at three sets of data now: 
1. My old, supposedly full, maintenance cartridge
2. A brand new genuine maintenance cartridge
3. The pastebin file

The pastebin file looks very similar to the brand new one.

I've also done some basic analysis of the contents of the used one and this corresponds with the contents of the others.

The 2048 bytes of the chip are organised as six distinct sections:

**Section 1 - 64 bytes**.
Bytes 0-1: not sure what these are (possibly checksum digits). 
Bytes 2-11: Next 10 bytes are the cartridge 9-character serial number (as little-endian 16-bit values) padded with a 0x00 byte.
Bytes 12-13: not sure what these are (possibly checksum digits)
Remaining bytes all 0x00

**Section 2 - duplicate of section 1**

**Section 3 - 320 bytes**
Bytes 0-1: 0xA5A5 in a brand new cartridge, followed by all 0x00s
Bytes 0-257: Hex data followed by 62 0x00s

**Section 4 - duplicate of section 3**

**Section 5 - 640 bytes**
Bytes 0-1: 0xA5A5 in a brand new cartridge, followed by all 0x00s
Bytes 0-513: Hex data followed by 126 0x00s

**Section 6 - duplicate of section 5**

Each section has checksum bytes (1st or last two bytes?) such that if you add contents as little-endian 16-bit numbers, each section sums to 0xA5A5, ignoring any overflow. 

The duplicate sections are - presumably - to mitigate against data corruption. There is exactly twice the amount of data in section 5 as in section 3, so I imagine these are tables logging something about each time ink is deposited in the cartridge. I also suspect that the first warning given to the user about replacing the cartridge is given when a certain number of entries is reached and the printer stops printing altogether when the tables are full or very close to full.

That's all I've got.

**Comment by gaby64** [comment]
This service mode eeprom report data might be usefull:

For full:
MTC D=100.0 MTC ST=2023/06/13-04:11
MTC SN(SN0=KNSX00215 SN1=  ) MTC CH=003 
MTC PC(M=009 R=012 S=001 T=001 D=000 C=001 I=005)

For empty:
MTC D=000.0 MTC ST=2023/10/13-06:29
MTC SN(SN0=KNSX00215 SN1= ) MTC CH=001 
MTC PC(M=000 R=000 S=000 T=000 D=000 C=000 I=000)

**Comment by gaby64** [comment]
> @gaby64 Does the rom in pastebin reset your old mc-g01 successfully?

The rom did successfully reset my old mc-g01

**Comment by Nickduino** [comment]
Can anyone *confirm* the hack works on the MC-G01 cartridge?

**Comment by wangyu-** [comment]
> Can anyone _confirm_ the hack works on the MC-G01 cartridge?

@gaby64 seems to have confirmed. But I am not sure if he was using the tool in this repo unchanged.

**Comment by gaby64** [comment]
> > Can anyone _confirm_ the hack works on the MC-G01 cartridge?
> 
> @gaby64 seems to have confirmed. But I am not sure if he was using the tool in this repo unchanged.

I used it unchanged, can confirm it works.

**Comment by MChamarinda** [comment]
> ![IMG_20231014_014054](https://private-user-images.githubusercontent.com/2108555/275143496-a621d0b3-0a18-47e3-8d98-f5a687c3bbe7.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQzODMxNTksIm5iZiI6MTcwNDM4Mjg1OSwicGF0aCI6Ii8yMTA4NTU1LzI3NTE0MzQ5Ni1hNjIxZDBiMy0wYTE4LTQ3ZTMtOGQ5OC1mNWE2ODdjM2JiZTcuanBnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDEwNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMDRUMTU0MDU5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTJiOTY1MDMwMWE3NjlkZWEzMzZlMzllMjcxYWMxMjI0MzNmYWRjZWVkY2NjZjJkNWJjMDA2ZDk4YTVmY2QxZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.ldU3hwZgKEZwRMsc62fvODgMuD0E_QUF__fV-ZOekbQ)

This works for me too❤️

**Comment by MChamarinda** [comment]
![Uploading IMG-20240104-WA0021.jpeg…]()

**Comment by MChamarinda** [comment]
Write the dump before cartridge fill.

**Comment by Nickduino** [comment]
> Look what someone was nice enough to post
> 
> https://pastebin.com/z24TQ36x

That someone is a really nice person!
I can confirm the sketch works with the MC-G01 and a good dump.

**Comment by Nickduino** [comment]
And I see they sell new sponge for the cartridge on Aliexpress if you don't want to make a mess cleaning it.

5 for 16.41€, that's cheaper than one cartridge from Canon:
https://www.aliexpress.com/item/1005005284132715.html

**Comment by raichea** [comment]
Having looked inside both a Canon cartridge and an AliExpress one, I don't think those pads will fit a genuine Canon maintenance cartridge.

**Comment by Nickduino** [comment]
Just to confirm...you looked inside a MC-G01 specifically?

**Comment by raichea** [comment]
Yes - I took it apart to have a go at cleaning the pads. Quite a messy process!

I took pictures but may not have them any more. I'll see if I can find them.

**Comment by Nickduino** [comment]
I hope you'll find your pics.

On this one, you see how they're supposed to fit:
![image](https://github.com/wangyu-/canon_mc-g02_resetter/assets/9784237/986fe173-33d2-422a-a48a-f33adfa7c581)

(Which means the top of the stack of the Aliexpress replacement sponge goes to the bottom of the cartridge)

**Comment by raichea** [comment]
You're in luck, I found the pictures in my computer's trash bin. 

After looking at the three cartridges as described in my post on 14 Oct 2023, I bought one of those Chinese versions and read the data on the chip (which is as expected but with a, presumably, random serial number encoded in the first two blocks of 64 bytes. I also dismantled it to see why the pads advertised on AliExpress looked so different to the ones I found in the genuine Canon cartridge. 

Internally the design is quite different. The Canon cartridge divides the internal space with plastic barriers and uses 12 unique custom-cut pads. I photographed them as I removed them for cleaning, labelling them with a marker pen to help with reassembly after cleaning and drying (some hope - the pads went all very dark after soaking and rinsing to remove the ink - but I was able to reassemble OK).

Anyway - here are the pictures of the Canon internals as I removed the pads:

![PXL_20231007_151438076](https://github.com/wangyu-/canon_mc-g02_resetter/assets/46376421/d36dfe18-024b-4dc0-82c3-0c889b57d197)
![PXL_20231007_151602119](https://github.com/wangyu-/canon_mc-g02_resetter/assets/46376421/a453c13c-e683-4c2b-8694-e227ad746482)
![PXL_20231007_151639585](https://github.com/wangyu-/canon_mc-g02_resetter/assets/46376421/554c8099-58fb-473b-b67d-717dd4bcfa95)
![PXL_20231007_151721143](https://github.com/wangyu-/canon_mc-g02_resetter/assets/46376421/cb608493-370e-4138-8798-23f71d3d1873)
![PXL_20231007_151824864](https://github.com/wangyu-/canon_mc-g02_resetter/assets/46376421/b15a87c4-3677-4339-9a97-9f6beed04019)

**Comment by Nickduino** [comment]
Thank you, that's very interesting. So the Chinese pads might work in a Chinese cartridge, but that's it.

**Comment by raichea** [comment]
I believe they're designed to fit the Chinese cartridges which all seem to be the same design (based on identical labelling on all versions I've found on the web).

**Comment by Nickduino** [comment]
> Look what someone was nice enough to post
> 
> https://pastebin.com/z24TQ36x

It looks like someone pasted the same but after one purge of the printer to compare the differences:
https://pastebin.com/7VTMcAKr

So it looks like you could erase all data, except for the serial number and strategically place a few 0xA5 values to end up with an "empty" cartridge.

But I wonder if the printer remembers the serial number of the last cartridge and throws out an error.

**Comment by Vrobenmat** [comment]
Does this still work for anyone? the pastebin dumps don't work for me, printer throws error 1725 'the maintenance cartridge cannot be recognised'. I've verified the ROM to make certain the data is correct. It looks like canon may be reading this as well and blocked these dumps in a firmware update.

**Comment by gaby64** [comment]
what printer model ?

**Comment by Vrobenmat** [comment]
> what printer model ?

GX7050

**Comment by Nickduino** [comment]
> It looks like canon may be reading this as well and blocked these dumps in a firmware update.

That would be such a dick move!

Have you read your EEPROM before reflashing it? Maybe you could use your original serial number and just clear the purging values

**Comment by Vrobenmat** [comment]
> > It looks like canon may be reading this as well and blocked these dumps in a firmware update.
> 
> That would be such a dick move!
> 
> Have you read your EEPROM before reflashing it? Maybe you could use your original serial number and just clear the purging values

Yeah, I read out the full cartridge, but it looks like it zaps the original structure - ended up with a block of 512 bytes repeated 4 times, and no serial number in sight. I have a new (3rd party) MC-G01 cartridge arriving today - will read it out before installing.

Pisses me off - the whole point of going the ink tank route was not to deal with cartridge crap anymore. I guess they still find a way to screw you on replacements. Happy to pay a higher base price for a printer that I can fix on the spot - why do none of these companies actually make a decent consumer/prosumer printer?

**Comment by bertybuttface** [comment]
@Vrobenmat I'm as pissed off as you are. I'm looking to update this code to make a simple ESP based device with the correct pinout that just clears these cartridges. If you can post another dump in addition to the pastebin above I'd really appreciate it. If I can compare a few I can write some code to just zero these out.

**Comment by Nickduino** [comment]
>  I have a new (3rd party) MC-G01 cartridge arriving today - will read it out before installing.

Did you read the virgin cartridge and again after a couple cycles to see which bits change?

**Comment by Vrobenmat** [comment]
I honestly threw it in without reading (because I needed to bloody print something...) Not used it that much since then, will pull it out and give it a read.

**Comment by hansipete** [comment]
> Look what someone was nice enough to post
> 
> https://pastebin.com/z24TQ36x

I copied that over to the [sketch_hack_write.ino](https://github.com/wangyu-/canon_mc-g02_resetter/blob/main/sketch_hack_write/sketch_hack_write.ino) and flashed my MC-G01 using a Seeduino Xiao. 

GX6050 is printing again — Thank you so much guys!

**Comment by bertybuttface** [comment]
> I honestly threw it in without reading (because I needed to bloody print something...) Not used it that much since then, will pull it out and give it a read.

I did the same thing, I have a used one and a partially used one I'll read out and post here. Hoping to work out how to generate a random serial number and zero the correct slots.

That way we won't need to share a ROM but the reading part won't be neccessary. Also should stop Canon blocking by serial number if they decide to do that (e.g. keeping a list of used serials on the printer or a blacklist).

**Comment by cweinberger** [comment]
Just wanted to confirm: worked amazingly with the ROM in this thread 👌

**Comment by CrisCard0so** [comment]
Can someone help me to share the dump ?

**Comment by CrisCard0so** [comment]
This is a dump from Canon G610. used ... :(
[24_16_20250911_173054.txt](https://github.com/user-attachments/files/22286381/24_16_20250911_173054.txt)

**Comment by Eric1797** [comment]
Anyone got the dump for MC G04 perhaps ?
just realised they soft block the printer when it is full i cant print any of the documents i even replaced the sponge didnt know the pesky chip wont let me print
